src/main/autoStart.ts

```diff
@@ -10,38 +10,51 @@ import { join } from "path";
 
 interface AutoStart {
     isEnabled(): boolean;
+    wasAutoStarted(): boolean;
     enable(): void;
     disable(): void;
 }
 
+const isFlatpak = process.env.FLATPAK_ID !== undefined;
+
 function makeAutoStartLinux(): AutoStart {
     const configDir = process.env.XDG_CONFIG_HOME || join(process.env.HOME!, ".config");
     const dir = join(configDir, "autostart");
     const file = join(dir, "vencord.desktop");
 
     return {
-        isEnabled: () => existsSync(file),
+        isEnabled: () => existsSync(file), // TODO: flatpak
+        wasAutoStarted: () => process.argv.includes("--autostart"),
         enable() {
-            const desktopFile = `
+            if (isFlatpak) {
+            } else {
+                const desktopFile = `
 [Desktop Entry]
 Type=Application
 Version=1.0
 Name=Vencord
 Comment=Vencord autostart script
-Exec=${process.execPath}
+Exec=${process.execPath} --autostart
```

Comment:
maybe this would be slightly more self explanatory
```suggestion
Exec=${process.execPath} --is-autostart=true
```